include/opencv.h

```diff
@@ -0,0 +1,183 @@
+/*******************************************************
+ * Copyright (c) 2016, ArrayFire
+ * All rights reserved.
+ *
+ * This file is distributed under 3-clause BSD license.
+ * The complete license agreement can be obtained at:
+ * http://arrayfire.com/licenses/BSD-3-Clause
+ ********************************************************/
+#pragma once
+#include <af/array.h>
+#include <af/defines.h>
+#include <af/traits.hpp>
+#include <af/exception.h>
+#include <vector>
+#include <opencv2/core/core.hpp>
+
+namespace afcv
+{
+
+/*
+ * opencv -> arrayfire
+ */
+
+/**
+    \ingroup opencv_interop
+    @{
+*/
+
+/**
+   This function will copy data from a cv::Mat to an \ref af::array
+
+   \param[in]  input The data which will be loaded into the array
+   \param[in]  transpose flag specifying if the input should be transposed during copy
+
+   \returns an af::array with the contents of the OpenCv Mat
+*/
+static af::array array(const cv::Mat& input, bool transpose = true)
+{
+    af::array output;
+    try {
+        int h = input[0].rows;
+        int w = input[0].cols;
+        int c = input[0].channels();
+        if (transpose) {
+            output = af::array(h, w, c, input.size());
+        } else {
+            output = af::array(w, h, c, input.size());
+        }
+        for (unsigned i = 0; i < input.size(); i++) {
+            if(input[i].channels() != c)
+                throw af::exception("number of OpenCV matrix channels inconsistent");
+            af::array tmp = array(input[i], transpose);
+            output(af::span, af::span, af::span, i) = tmp;
+        }
+        return output;
+    } catch (const af::exception& ex) {
+        throw af::exception("std::vector<cv::Mat> to af::array error ");
```

Comment:
I believe this error message belongs in the other `array()` function and vice-versa.